apps/crn-server/test/data-providers/contentful/analytics.data-provider.test.ts

```diff
@@ -0,0 +1,1365 @@
+            [
+              {
+                sys: {
+                  id: 'interest-group-id-1',
+                },
+              },
+            ];
+          contentfulGraphQLResponse.teamsCollection!.items[0]!.linkedFrom!.teamMembershipCollection!.items[0]!.linkedFrom!.usersCollection!.items.push(
+            {
+              alumniSinceDate: pastDate,
+              linkedFrom: {
+                interestGroupLeadersCollection: {
+                  items: [
+                    {
+                      linkedFrom: {
+                        interestGroupsCollection: {
+                          items: [
+                            {
+                              sys: {
+                                id: 'interest-group-id-1',
+                              },
+                            },
+                          ],
+                        },
+                      },
+                    },
+                    {
+                      linkedFrom: {
+                        interestGroupsCollection: {
+                          items: [
+                            {
+                              sys: {
+                                id: 'interest-group-id-2',
+                              },
+                            },
+                          ],
+                        },
+                      },
+                    },
+                  ],
+                },
+              },
+            },
+          );
+          contentfulGraphqlClientMock.request.mockResolvedValueOnce(
+            contentfulGraphQLResponse,
+          );
+
+          const result = await analyticsDataProvider.fetchTeamLeaderShip({});
+
+          expect(
+            result.items[0]!.interestGroupPreviousLeadershipRoleCount,
+          ).toBe(2);
+        });
+      });
+
+      describe('Interest Group Member Count', () => {
+        test('Should return 0 for interestGroupMemberCount when the client returns interestGroupsCollection items as an empty array', async () => {
+          const contentfulGraphQLResponse = getAnalyticsTeamLeadershipQuery();
+          contentfulGraphQLResponse.teamsCollection!.items[0]!.linkedFrom!.interestGroupsCollection!.total = 0;
+          contentfulGraphqlClientMock.request.mockResolvedValueOnce(
+            contentfulGraphQLResponse,
+          );
+
+          const result = await analyticsDataProvider.fetchTeamLeaderShip({});
+
+          expect(result.items[0]!.interestGroupMemberCount).toBe(0);
+        });
+
+        test('Should return interestGroupMemberCount of 2 when the client returns a total of 2 interest groups associated with a team', async () => {
+          const contentfulGraphQLResponse = getAnalyticsTeamLeadershipQuery();
+          contentfulGraphQLResponse.teamsCollection!.items[0]!.linkedFrom!.interestGroupsCollection!.total = 2;
```

Comment:
what if the interest group is inactive?

Reply:
good point, might need to raise this with Joao actually